### PR TITLE
Retry cloning VM for all available hosts in case of host failure

### PR DESCRIPTION
**Pull request by jablonskim**

### CODE REVIEW

File: ansible/files/provision_vm.py [hunk 1]
```diff
@@ -125,23 +122,34 @@ def provision_vm(api, args):
...
-        task = template.Clone(name=name, folder=folder, spec=clone_spec)
-        WaitForTask(task)
-    except (KeyboardInterrupt, SystemExit):
-        if task is not None:
-            # In this case we should at least try to cancel the task
-            task.CancelTask()
-            wait_until_vm_does_not_exist(api, name)
-        raise
+
+    hosts_and_datastores = api.get_destination_hosts_and_datastores(args.datastore_cluster)
```
Comment (jacek-izykowski) [inline ansible/files/provision_vm.py]: Can you, please, extract a function here (e.g. lines 126 - 146)?

File: ansible/files/provision_vm.py [hunk 2]
```diff
@@ -125,23 +122,34 @@ def provision_vm(api, args):
...
+            if task is not None:
+                # In this case we should at least try to cancel the task
+                task.CancelTask()
+                wait_until_vm_does_not_exist(api, name)
+            raise
+        except vim.fault.InvalidHostState:
+            # TODO: Post a warning about cloning failure to some monitoring service
+            continue
+
+    raise ResourceNotFound('Choosing appropriate host and datastore failed')
```
Comment (jacek-izykowski) [inline ansible/files/provision_vm.py]: Can you find a more suitable exception class here (or create a new one)? What do you think?
Can we return all messages from previously encountered InvalidHostState exceptions here?

File: ansible/files/vmware_api.py [hunk 3]
```diff
@@ -74,18 +74,13 @@ def get_vm_folder(self, folder_path):
         return self.content.searchIndex.FindByInventoryPath(inventory_path)
 
 
-    def select_destination_host_and_datastore(self, datastore_cluster_name):
+    def get_destination_hosts_and_datastores(self, datastore_cluster_name):
```
Comment (jacek-izykowski) [inline ansible/files/vmware_api.py]: MKos: Maybe a better name that suggests that this is actually generator?

**Comment by jablonskim** [comment]
recheck no bug

**Pull request merged**